libvirt/resource_libvirt_domain.go

```diff
@@ -344,16 +346,31 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 			if err != nil {
 				return fmt.Errorf("Error retrieving network name: %s", err)
 			}
+			dhcpMutex.Lock()
 			networkDef, err := newDefNetworkfromLibvirt(&network)
+			dhcpMutex.Unlock()
```

Comment:
Yes, in my testing I had a loop that would create a network of each type and 2 VMs with a NIC on each network, and without this locking it would fail very quickly, whereas with locking I couldn't get it to fail. However, now that I look at it again, I can see the following case:

1. Thread1 acquires lock, reads network1 (no DHCP range), releases lock
2. Thread2 acquires lock, reads network1 (no DHCP range), releases lock
3. Thread 1 advances to enableDHCP, acquires lock, sees no DHCP, enables DHCP, releases lock
4. Thread 2 advances to enableDHCP, acquires lock, sees no DHCP, enabled DHCP -> ERROR

It seems we might be better to just *not* lock here, then read the libvirt network *again* inside the locking of enableDHCP. I think that might be best, because this locking seems to just protect against the case where Thread 1 is currently enabling DHCP while Thread 2 is a bit further behind and is trying to read the network.